nixos/modules/services/misc/jellyfin.nix

```
@@ -16,6 +16,11 @@ in
         description = "User account under which Jellyfin runs.";
       };
 
+      package = mkDefault (
+        if versionAtLeast config.system.stateVersion "20.09" then pkgs.jellyfin
+        else pkgs.jellyfin_10_5
+      );
```

```suggestion
      package = mkOption {
        type = types.package;
        example = literalExample "pkgs.jellyfin";
        description = ''
          Jellyfin package to use.
        '';
      };
```

then you can add this to the configuration part:

```nix
    services.jellyfin.package =
      mkDefault (if versionAtLeast config.system.stateVersion "20.09" then pkgs.jellyfin
        else pkgs.jellyfin_10_5);
```

and finally, you can replace `pkgs.jellyfin` by `cfg.package` in the `ExecStart` line of the configuration part.